internal/framework/status/statuses.go

```diff
@@ -40,7 +40,7 @@ func (n *NginxGatewayStatus) APIGroup() string {
 }
 
 // ListenerStatuses holds the statuses of listeners where the key is the name of a listener in the Gateway resource.
-type ListenerStatuses map[string]ListenerStatus
+type ListenerStatuses []ListenerStatus
```

Comment:
[1] This PR changes the type of `GatewayStatus.ListenerStatuses` from map to slice.